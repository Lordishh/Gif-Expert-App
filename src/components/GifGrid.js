import React from "react";
import useFetchGifs from "../hooks/useFetchGifs";
import PropTypes from "prop-types";

import GifGridItem from "./GifGridItem";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="title animate__animated animate__rubberBand">
        {" "}
        Gifs de la b√∫squeda de "{category}".
      </h3>

      {loading && (
        <p className="animate__animated animate__flash loading">Cargando...</p>
      )}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  categoty: PropTypes.string.isRequired,
};

export default GifGrid;
