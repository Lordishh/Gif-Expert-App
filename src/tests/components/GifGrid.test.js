import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";

import GifGrid from "../../components/GifGrid";
/* import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs"); */

describe("Pruebas en GifGrid", () => {
  const category = "One Punch";
  /*   test("debe de mostrarse correctamente", () => {
    const gifs = [
      {
        id: "ABC",
        url: "https://localhost/cualquier/cosa.jpg",
        title: "cualquier cosa",
      },
    ];
    useFetchGifs.mockReturn({
      data: gifs,
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  }); */

  test("debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs", () => {
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("p").exists()).toBe(false);
  });
});
