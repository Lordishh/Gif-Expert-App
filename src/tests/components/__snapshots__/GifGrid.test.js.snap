// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGrid debe de mostrar items cuando se cargan imágenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="title animate__animated animate__rubberBand"
  >
     
    Gifs de la búsqueda de "
    One Punch
    ".
  </h3>
  <p
    className="animate__animated animate__flash loading"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
